forte/processors/data_augment/algorithms/embedding_similarity_replacement_op.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+
+from typing import Tuple
+import numpy as np
+
+from ft.onto.base_ontology import Annotation
+from forte.common.configuration import Config
+from forte.processors.data_augment.algorithms.text_replacement_op \
+    import TextReplacementOp
+
+__all__ = [
+    "EmbeddingSimilarityReplacementOp",
+]
+
+
+class EmbeddingSimilarityReplacementOp(TextReplacementOp):
+    r"""
+    This class is a replacement op leveraging pre-trained word
+    embeddings, such as word2vec and glove, to replace the input
+    word with another word with similar word embedding.
+    By default, the replacement word is randomly chosen from the
+    top k words with the most similar embeddings.
+
+    Args:
+        configs:
+            The config should contain the following key-value pairs:
+            embedding: A texar.tf.data.Embedding object. Can be initialized
+                from pre-trained embedding file
+            vocab: A texar.torch.data.Vocab object. Can be initialized from
+                a vocabulary file
+            top_k: the number of k most similar words to choose from
+    """
+    def __init__(self, configs: Config):
```

Comment:
I think about this parameter set-up a little bit more. Now, this stores whole python objects into a Config. I think this in the long term will hurt the system. Configs are normally not used this way. I think even sending the objects directly is better than this method. Maybe we should simply do the following?

```suggestion
    def __init__(self, embedding: Embedding, vocab: Vocab):
```

Reply:
I agree. In fact the code was changed from the latter format.

But currently, the `ReplacementDataAugmentProcessor` requires a `data_aug_op_config` for it to work. It needs to initialize the data_aug_op from that Config object.

Maybe we can instead pass the parameters to initialize the `Embedding` and `Vocab` objects, then initialize those objects in this `__init__` function?

Reply:
We can still pass the required `data_aug_op_config` to fit the class requirement, but use handle our own Embeddings here?

Reply:
Okay. I updated the Config parameters